refactor: async fetcher (#29)
Agile-Arch-Angels/decentralized-tech-radar_dev: docs/dev_docs/fetcher.md

# Fetcher

> The fetcher is used to pull whitelisted files/folders from one or more git repositories. The progress of the fetcher can be tracked using the small spinner and progress bar displayed in the terminal. It takes a string containing 3 values:  

Author: Currently the whitelist can contain folders. The git folder will pull all files from that folder going down. The program will then recursively search those folders for CSV files so the original text is correct

> 1. A URL to a git based repository
> 2. A branch name
> 3. A path to a whitelist file  

Reviewer: ```suggestion
1. An URL to a git-based repository
2. A branch name
3. A path to a whitelisted file
```

Reviewer 2: It is actually 🤓 "A URL" not "An URL"

Reviewer: Um actually, I concede 🤓

Author: 3. is wrong. It is a path to a whitelist file. The whitelist file contains whitelisted files or folders

func FetchFiles(url, branch, specFile string) error 
```

### Asynchronous

> The fetcher runs using go functions which means it runs asynchronously. This allows the fetcher to fetch from multiple repositories at the same time greatly decreasing the total running time on fetch calls with many repositories.

Reviewer: ```suggestion
The fetcher runs using Go functions, meaning it runs asynchronously. This allows simultaneous fetching of files from multiple repositories, greatly decreasing the total running time on fetch calls with many repositories.
```
